cpu/em_cpuhotplug.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2017 IBM
+# Author: Gautham R. Shenoy <ego@linux.vnet.ibm.com>
+# Author: Shriya Kulkarni <shriyak@linux.vnet.ibm.com>
+import random
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import process, cpu
+
+
+class cpuhotplug_test(Test):
+    """
+    To test hotplug within core in random manner.
+    """
+    def setUp(self):
+        """
+        Get the number of cores and threads per core
+        Set the SMT value to 4/8
+        """
+        if 'ppc' not in platform.processor():
+            self.cancel("Processor is not ppc64")
+        self.nfail = 0
+        self.CORES = process.system_output("lscpu | grep 'Core(s) per socket:'"
+                                           "| awk '{print $4}'", shell=True)
+        self.SOCKETS = process.system_output("lscpu | grep 'Socket(s):'"
+                                             "| awk '{print $2}'", shell=True)
+        self.THREADS = process.system_output("lscpu | grep 'Thread(s) per core"
+                                             ":'| awk '{print $4}'",
+                                             shell=True)
+        self.T_CORES = int(self.CORES) * int(self.SOCKETS)
+        self.log.info(" Cores = %s and threads = %s "
+                      % (self.T_CORES, self.THREADS))
+        process.system("echo 8 > /proc/sys/kernel/printk", shell=True)
+        self.max_smt = 4
+        if cpu.get_cpu_arch().lower() == 'power8':
+            self.max_smt = 8
+        if cpu.get_cpu_arch().lower() == 'power6':
+            self.max_smt = 2
+        process.system_output("ppc64_cpu --smt=%s" % self.max_smt, shell=True)
+        self.path = "/sys/devices/system/cpu"
+
+    def test(self):
+        """
+        This script picks a random core and then offlines all its threads
+        in a random order and onlines all its threads in a random order.
+        """
+        for x in range(1, 100):
+            self.log.info("================= TEST %s ==================" % x)
+            core_list = []
+            core_list = self.random_gen_cores()
+            for core in core_list:
```

Comment:
can it be 
`for core in self.random_gen_cores()`

Reply:
What you need is a list of shuffled random cpus between 1 to 100 ?
core_list = [ 2, 99, 34, 53, 56, 22, 12. ......]

the self.random_gen_cores() returns a list so no need to initialize core_list = []

Reply:
Yes, @abdhaleegit @PraveenPenguin  : I will make this change